src/app/child-dev-project/reporting/aggregation.service.spec.ts

```diff
@@ -0,0 +1,253 @@
+import { TestBed } from "@angular/core/testing";
+
+import { AggregationService } from "./aggregation.service";
+import { EntityMapperService } from "../../core/entity/entity-mapper.service";
+import { Child } from "../children/model/child";
+import { Gender } from "../children/model/Gender";
+import { School } from "../schools/model/school";
+import { RecurringActivity } from "../attendance/model/recurring-activity";
+import { EventNote } from "../attendance/model/event-note";
+import { Note } from "../notes/model/note";
+import moment from "moment";
+
+describe("AggregationService", () => {
+  let service: AggregationService;
+  const mockEntityMapper: jasmine.SpyObj<EntityMapperService> = jasmine.createSpyObj(
+    ["loadType"]
+  );
+
+  const loadTypeFake = (
```

Comment:
If you have complex test data, using MockDatabase may be easier than directly mocking EntityMapperService.

`MockDatabase.createWithData(arrayOfEntities)` let's you init the in-memory database with any entities and let EntityMapper handle the filtering based on types, etc.